### DISCUSSION TITLE
Hi, I try to do unit test with miniredis v2 library, but it doesn't work.

### DISCUSSION THREAD

**Discussion by ghost** [redis/rueidis#216] [Q&A] [answered]
I am trying to perform unit tests using the miniredis (https://github.com/alicebob/miniredis) library. During the unit tests, a panic occurs related to the MGet operation, stating "multi key command with different key slots are not allowed". Upon debugging, I found out that the Redis client is operating in cluster mode. The code that causes the error is as follows:
`d, err := newForRedisTest(mr.Addr())`
In this case, `mr.Addr()` passes only one host value (127.0.0.1:52234). Despite the rueidis library's README stating that it operates in single client mode when there is only one host, I confirmed that a cluster client is created. However, when connecting to the Redis server on my local machine (:6379) instead of the mock Redis client, I confirmed that it is created as a single client. What could be causing this difference?

Additionally, I would like to make a suggestion. Currently, the client is created in cluster mode, and if it fails, it operates as a single client. However, this code may cause unexpected issues, such as the one I mentioned earlier where it operates in cluster mode when I want it to operate in single client mode. Therefore, I would appreciate it if the `type ClientOption struct` included a `SingleClientMode bool` field, allowing developers to explicitly use a Single Client.

**Comment by rueian** [comment, marked as answer]
Hi @KwonsooPark,

rueidis will send CLUSTER SLOTS commands to check if you are connecting to a cluster enabled node. If that command fails then rueidis will switch to single client .
